scripts/rake.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+__author__ = "Øyvind Jekteberg and Kristian Gingstad"
+__copyright__ = "Copyright 2020, The ArXivDigest Project"
+
+import nltk
+import string
+from nltk.tokenize import word_tokenize
+from collections import Counter, defaultdict
+import re
+from nltk.util import ngrams
+from enum import Enum
+
+
+class Metric(Enum):
+    """Different metrics that can be used for ranking candidate keywords."""
+    DEGREE_TO_FREQUENCY_RATIO = 0  
+    WORD_DEGREE = 1 
+    WORD_FREQUENCY = 2  
+
+
+class Rake(object):
+    def __init__(self, max_length, scoring_metric=Metric.DEGREE_TO_FREQUENCY_RATIO, stopwords=None, punctuation=None):
+        """Initializes a rake object with english stop words and string punctuations.
+        Must define max_lenth and what scoring metric to use.
+        Scoring metrics available are f, d, df."""
+        self.stopwords = stopwords if stopwords else nltk.corpus.stopwords.words('english')
+        self.punctuations = punctuation if punctuation else string.punctuation.replace('-', '')
+        self.max_length = max_length
+        self.scoring_metric = scoring_metric
+        self.frequency = None
+        self.degree = None
+        self.rank_list = None 
+        self.ranked_phrases = None
+
+    def extract_keyword_from_sentences(self, sentences): 
+        """Extracts keywords from list of sentences."""
+        phrase_list = Counter()  #TODO prob not set
+        for sentence in sentences:
+            word_list = [word.lower() for word in word_tokenize(sentence)]
+            phrase_list = phrase_list + Counter(self.get_candidate_keywords(word_list))
+        phrase_list = phrase_list + Counter(self.find_adjoining_keywords(sentences))
+        self.frequency = self.create_freq_dist(phrase_list)
+        self.degree = self.create_degree_dist(phrase_list)
+        self.find_ranked_keywords(phrase_list)
+
+    def get_candidate_keywords(self, tokenized_sentence):
```

Comment:
All methods that are not supposed to be called from outside the class should be marked as private by prefixing them with double underscores. (This should be a general convention for other classes as well)